### PR TITLE
Fix unexported field in collateKvRecursive

### PR DESCRIPTION
**Pull request by cocap10**
This PR :
 - Fixes unexported field in collateKvRecursive
 - Tests the fix
 - Updates readme

### CODE REVIEW

**Comment by cocap10** [comment]
@emilevauge : Can you have a look at it ?

File: README.md [hunk 1]
```diff
@@ -7,7 +7,8 @@ Stært is a Go library for loading and merging a program configuration structure
 
 ## Overview
 Stært was born in order to merge two sources of Configuration ([Flæg](https://github.com/containous/flaeg), [Toml](http://github.com/BurntSushi/toml)).
-We developed [`flaeg`](https://github.com/containous/flaeg) and `staert` in order to simplify configuration maintenance on [traefik](https://github.com/containous/traefik).
+Now it supports also [Key-Value Store](#kvstore).
```
Comment (emilevauge) [inline README.md]: also supports

File: README.md [hunk 2]
```diff
@@ -186,7 +187,7 @@ It handles :
 Note : Hopefully, we provide the function `StoreConfig` to store your configuration structure ;)
 
 ### KvSource
-KvSource impement Source: 
+KvSource impements Source:
```
Comment (emilevauge) [inline README.md]: implements

**Comment by emilevauge** [comment]
LGTM, once typos corrected

**Pull request merged**